src/player.cpp

```diff
@@ -702,8 +702,9 @@ uint16_t Player::getLookCorpse() const
 	return ITEM_MALE_CORPSE;
 }
 
-void Player::addStorageValue(const uint32_t key, const int32_t value, const bool isLogin /* = false*/)
+void Player::addStorageValue(const uint32_t key, const std::optional<int32_t> data, const bool isLogin /* = false*/)
```

Comment:
The value should not be optional here. If the user wants to remove the value, they should use another function `removeStorageValue`. `add` should always set to something.

Reply:
I like the idea, I'll try to refactor it, and add the caveat you mentioned